### DISCUSSION TITLE
How to set background image to iText in fabricjs

### DISCUSSION THREAD

**Discussion by rohitob** [fabricjs/fabric.js#7297] [Q&A]
I am working with `fabricjs canvas`.
I want to set background image to the `iText`.
I know `fill` option, but it's filling to the text so don't want to fill text, I want text's background as shown in below image.

![iText_bg_img](https://user-images.githubusercontent.com/80304141/129155114-17ce252a-8ccd-4250-bc66-134ca8570065.png)

As we can see in image, **I want to add text in canvas like this.**

Thank you!

[https://stackoverflow.com/questions/68752546/set-background-image-to-itext-in-fabricjs](https://stackoverflow.com/questions/68752546/set-background-image-to-itext-in-fabricjs)

## Information about environment
Typescript

**Comment by asturur** [comment]
the object background color, as of now takes only a color.
It could take a pattern or a gradient in theory but for not is not supported.

**Reply by rohitob** [reply]
Okay
Thank you @asturur
